playground/test_playground.sh

```diff
@@ -40,9 +41,59 @@ buyAmount=$(jq -r --args '.quote.buyAmount' <<< "${quote_reponse}")
 feeAmount=$(jq -r --args '.quote.feeAmount' <<< "${quote_reponse}")
 #echo -e $sellAmount"\n"$buyAmount"\n"$feeAmount
 validTo=$(($(date +%s) + 120)) # validity time: now + 2 minutes
+#order="0x"$(printf '%s' "$quote_reponse" | hexdump -ve '/1 "%02X"') #"$quote_reponse #(jq -r --args '.quote' <<< "${quote_reponse}")
+
+echo $quote_reponse
+
+# Filter out unneeded fields
+order_proposal=$(jq -r --args '.quote|=(.appData=.appDataHash) | del(.quote.appDataHash, .quote.signingScheme) | .quote' <<< "${quote_reponse}")
+
+# Prepare EIP-712 typed struct
+eip712_typed_struct='{
+  "types": {
+    "EIP712Domain": [
+      { "name": "name", "type": "string" },
+      { "name": "version", "type": "string" },
+      { "name": "chainId", "type": "uint256" },
+      { "name": "verifyingContract", "type": "address" }
+    ],
+    "Order": [
+      { "name": "sellToken", "type": "address" },
+      { "name": "buyToken", "type": "address" },
+      { "name": "receiver", "type": "address" },
+      { "name": "sellAmount", "type": "uint256" },
+      { "name": "buyAmount", "type": "uint256" },
+      { "name": "validTo", "type": "uint32" },
+      { "name": "appData", "type": "bytes32" },
+      { "name": "feeAmount", "type": "uint256" },
+      { "name": "kind", "type": "string" },
+      { "name": "partiallyFillable", "type": "bool" },
+      { "name": "sellTokenBalance", "type": "string" },
+      { "name": "buyTokenBalance", "type": "string" }
+      ]
+    },
+  "primaryType": "Order",
+  "domain": {
+    "name": "Gnosis Protocol",
+    "version": "v2",
+    "chainId": 100,
```

Comment:
chain id belongs to `gnosis chain` whereas the test is probably supposed to happen on mainnet.